### DISCUSSION TITLE
How to use UseQuery to assign value to a pre-defined variable?

### DISCUSSION THREAD

**Discussion by heartcoreapex** [TanStack/query#6395] [Q&A]
Hi I'm a vue3 & vue-query newbie
I understand the normal assignment would be 

`const { data: name_of_your_variable } = UseQuery({});`

but suppose I have a pre-existing variable

I don't think I can do the following
```
const name_of_my_variable
const { data: name_of_my_variable} = UseQuery({});
```

I'm looking to change the value of the variable via 
`name_of_my_variable.value = `

Suppose the following scenario
```
quote = ref({
    ID,
    currencyRate
});

const { data: currencyRate } = useQuery({
	queryKey: ['currencyRate', { currencyID, effectiveDate }],
	queryFn: async () =>  {
		const response = await http.get('rate/currency', { params: { CurrencyID: currencyID, EffectiveDate: effectiveDate } });
		return response.data;
	},
});
```
How do I assign the currencyRate grabbed from useQuery to quote.value.currencyRate?

**Comment by oren-legit** [comment]
Why would you need this?

**Reply by heartcoreapex** [reply]
cause I need to assign the value to a particular nested object I have

**Reply by oren-legit** [reply]
I think its better to rethink your data flow so that the nested object is derived from what's returned from the query

**Reply by heartcoreapex** [reply]
is there some eventhandler or lifecycle hook I should be aware of in vue3?
b/c onSuccess is deprecated...

**Reply by heartcoreapex** [reply]
updated the original post
